Sources/TensorFlow/Layers/Embedding.swift

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// An input structure containing the embedding indices.
+///
+/// - Note: Often times, `Embedding` is followed by a `Flatten` and a `Dense` layer. When this 
+///   is the case, ensure that all input sequences of indices have the same dimension.
+// NOTE: This structure is needed to conform `Embedding` to the Layer protocol.
+@frozen
+public struct EmbeddingInput: Differentiable {
+    /// Sequences of indices that will be passed into the layer.
+    @noDerivative public var indices: Tensor<Int32>
```

Comment:
It's absolutely possible to have an embedding layer with more than 2^32 embeddings (and therefore one that requires `Int64` indices), but it's fine to not worry about that for now.

Reply:
`gathering(atIndices:alongAxis:)` is not yet generic over `BinaryInteger`. To fix this, we need to start from there.

Reply:
https://github.com/tensorflow/swift-apis/issues/259